halo2_proofs/src/plonk/circuit.rs

```diff
@@ -817,6 +888,50 @@ pub enum Expression<F> {
     Scaled(Box<Expression<F>>, F),
 }
 
+impl<F> Into<ExpressionMid<F>> for Expression<F> {
```

Comment:
I noticed this; the reason why I implemented `Into` and not `From` is because I was thinking about the crate split.  It would look like this:
- halo2_middleware
  - ExpressionMid
- halo2_backend (imports halo2_middleware)
  - Function to convert ExpressionMid to Expression

In this setting, we can't implement `From`, only `Into`!  I think there are many details like this that will look more clear (and also will need to be addressed) after the crate split.

Another point of discussion is whether doing the crate split in this PR or in a future PR.  I personally would prefer doing it in this PR, but both options seem fine to me.

Reply:
I think you can still implement `From` in this setting though. 

Also, agree on doing the split on this PR.